admin-portal/src/pages/editPledge/components/pledgeSummary/pledgeSummary.tsx

```diff
@@ -21,6 +24,46 @@ export default function PledgeSummary({
     }
   };
 
+  useEffect(() => {
+    getPledgeStatus(aidStatus);
+  }, []);
+
+  const [pledgeActive, setPledgeActice] = useState<boolean>();
+  const [paymentInitatedActive, setPatmetInitiatedActive] = useState<boolean>();
+  const [paymentComfirm, setPaymentConfirActive] = useState<boolean>();
+
+  function getPledgeStatus(status: AidPackage.Status) {
```

Comment:
I think we can solve this easily by keeping a map something like this:

const statusToAllowedPackageStates = {
       Pledge.Status.Published: [AidPackage.Status.Draft, AidPackage.Status.Published, ...],
       ....

}